app/javascript/tests/patient/monitoring_actions/AssignedUser.test.js

```diff
@@ -76,44 +76,44 @@ describe('AssignedUser', () => {
     const wrapper = mount(<AssignedUser patient={mockPatient1} assigned_users={assigned_users} current_user={mockUser1} jurisdiction_paths={mockJurisdictionPaths} authenticity_token={mockToken} household_members={[mockPatient2, mockPatient3, mockPatient4]} workflow={'global'} />);
     wrapper
       .find('#assigned_user')
-      .at(1)
+      .hostNodes()
       .simulate('change', { target: { id: 'assigned_user', value: '1' } });
     wrapper.find(Button).simulate('click');
 
     // initial radio button state
-    expect(wrapper.find(ApplyToHousehold).exists()).toBeTruthy();
-    expect(wrapper.find(CustomTable).exists()).toBeFalsy();
-    expect(wrapper.state('apply_to_household')).toBeFalsy();
-    expect(wrapper.find('#apply_to_household_no').at(1).prop('checked')).toBeTruthy();
-    expect(wrapper.find('#apply_to_household_yes').at(1).prop('checked')).toBeFalsy();
+    expect(wrapper.find(ApplyToHousehold).exists()).toBe(true);
+    expect(wrapper.find(CustomTable).exists()).toBe(false);
+    expect(wrapper.state('apply_to_household')).toBe(false);
+    expect(wrapper.find('#apply_to_household_no').hostNodes().prop('checked')).toBe(true);
```

Comment:
I'm not clear on all the changes from `at(index)` to `hostNodes()`. Just going off what I read in the docs, `at` returns a wrapper around a single node (the node at the specified index) and `hostNodes` returns a wrapper around a collection of nodes (the nodes that are HTML elements). Can you explain the reason for those changes and why it works?

Reply:
For the most part in our tests we use `shallow`, which is meant to test components in isolation and does not render child components.  However we have to use `mount` in the occasional case when we need the full DOM rendering, which is needed here because we need the `CustomTable` to render.  When you use mount though, it fully renders ALL children, this includes inputs/buttons/etc.  The annoying result of this is that there is a `<Form.Control>` and an `<input>` with the same id, whereas if you shallow render there is just  a `<Form.Control>`. This means when you do a `find(#some_id)` to select an input or button, two items are returned when you mount.  Yes you could use `at(0)` to just select the first one, but the preferred method is to use `hostNodes()`.  That way if we did swap back to shallow, the test theoretically would still pass because it would still grab the `<Form.Control>`.

Reply:
Ah thank you for explaining all of that.